src/codegen/mod.rs

```diff
@@ -7,6 +7,8 @@ use cu_ast as cu;
 use std::collections::HashMap;
 use std::iter;
 
+static mut LABEL_COUNTER: i32 = 0; 
```

Comment:
There is the function `utils::fresh_name`. Would it be sensible to use that instead of the counter?